paasta_tools/autoscaling_lib.py

```diff
@@ -610,9 +611,8 @@ def wait_and_terminate(slave, dry_run):
                 if not instance_id:
                     log.warning("Didn't find instance ID for slave: {0}. Skipping terminating".format(slave['pid']))
                     continue
-                is_ready_to_kill = True
-                # is_ready_to_kill = paasta_maintenance.something(slave['pid'])
-                if is_ready_to_kill:
+                # Check if no tasks are running or we have reached the maintenance window
+                if is_safe_to_kill(slave['hostname']):
                     log.info("TERMINATING: {0}".format(instance_id))
```

Comment:
We probably want to log the hostname/ip as well (since going from an instance_id to a hostname/ip after the instance has been terminated can be challenging).